fsspec_reference_maker/combine_v2.py

```diff
@@ -0,0 +1,291 @@
+            if coo_attrs:
+                arr.attrs.update(coo_attrs.get(k, {}))
+            arr.attrs["_ARRAY_DIMENSIONS"] = [k]
+
+    def second_pass(self):
+        for i, fs in enumerate(self.fss):
+            m = fs.get_mapper("")
+
+            for v in fs.ls("", detail=False):
+                cvalues = {c: self._get_value(i, fs, c, arrname=v, fn=self._paths[i])
+                           for c in self.coo_map}
+                var = cvalues.pop("var")
+                var = var[0] if isinstance(var, list) else var
+                zarray = ujson.loads(m[f"{v}/.zarray"])
+                zattrs = ujson.loads(m[f"{v}/.zattrs"])
+                coords = zattrs["_ARRAY_DIMENSIONS"]
+                coord_order = self.concat_dims + [c for c in coords if c not in self.concat_dims]
+                for fn in fs.ls(var, detail=False):
+                    if ".z" in fn:
+                        continue
+                    key_parts = fn.split("/", 1)[1].split(".")
+                    key = f"{var}/"
+                    for c in coord_order:
+                        if c in self.coos:
+                            i = np.searchsorted(self.coos[c], cvalues[c])
+                            key += str(i // self.chunks[c])
+                        else:
+                            key += key_parts.pop(0)
+                        key += "."
+                    key = key.rstrip(".")
+                    self.out[key] = fs.references[fn]
+
+                if f"{var}/.zarray" not in self.out:
+                    for k in reversed(self.concat_dims):
+                        shape = zarray["shape"]
+                        chunks = zarray["chunks"]
+                        coords.insert(0, k)
+                        shape.insert(0, self.coos[k].size)
+                        chunks.insert(0, self.chunks[k])
+                    self.out[f"{var}/.zarray"] = ujson.dumps(zarray)
+                    self.out[f"{var}/.zattrs"] = ujson.dumps(zattrs)
+
+    def consolidate(self, inline_threshold=500):
+        """Turn raw references into output"""
+        out = {}
+        for k, v in self.out.items():
+            if isinstance(v, list) and v[2] < inline_threshold:
+                v = self.fs.cat_file(v[0], start=v[1], end=v[1] + v[2])
+            if isinstance(v, bytes):
+                try:
+                    # easiest way to test if data is ascii
+                    out[k] = v.decode('ascii')
+                except UnicodeDecodeError:
+                    out[k] = (b"base64:" + base64.b64encode(v)).decode()
+            else:
+                out[k] = v
+        return {"version": 1, "refs": out}
+
+
+def example_dso():
+    """Special dataset has same offset/size in every single file
+
+    We get concatenation coordinates purely from the file names; we align wavelength images,
+    which were taken in sequence, to the nearest instance of the shortest wavelength.
+    """
+    import fsspec_reference_maker.fits
```

Comment:
Is `fits()` (now `kerchunk.fits()`?) a function that is TBD?

Reply:
It's in this PR: https://github.com/fsspec/kerchunk/pull/83/files#diff-e2d10fd56b4a80740b843f8bc0752225f72c82713b5c1364785613058fa68b3b

The SDO example doesn't really belong here, though, should be removed (along with any of the WCS stuff in fits.py, probably).